### ISSUE TITLE
Creative Commons license versions

### ISSUE THREAD

**Issue by Maniues** [Chocobozzz/PeerTube#4633]
Dear PeerTube community,

Some videos are licensed under CC Licenses (Attribution, non-commercial etc.). You can choose CC License, but there are many versions of that licenses (1.0,2.0,2.5,3.0,4.0).  PeerTube doesn't allow to select for example 3.0 version, but actually, PeerTube redirects users to CC Beta Chooser, so the best option is to add 4.0 versions (and CC0v1).
Solution:
Add versions and URLs to that licenses

**Comment by Poslovitch** [comment]
Working on this :slightly_smiling_face:

**Comment by Poslovitch** [comment]
How would we go with adding a link?
![image](https://user-images.githubusercontent.com/20014332/145812556-a790516b-dfbb-47ef-a632-bf2963537551.png)

Licences names are quite long already.

**Comment by Maniues** [comment]
Just add "4.0", without acronyms, so:
- Attribution - Share Alike 4.0 
instead of
- Attribution - Share Alike (CC-BY-SA 4.0)

**Comment by Maniues** [comment]
Or just add URLs to these licenses and omit the "4.0" or acronyms.
I generated random movie from peertube website: https://vidcommons.org/w/9952af68-0b30-4ea0-81ac-a26444541331
Look:
![obraz](https://user-images.githubusercontent.com/66561397/145818492-89479603-9920-4132-b544-7793ff26ec78.png)
If you click on the license name, you can search videos under that license. I propose change it to license URL on creativecommons.org

**Comment by frankstrater** [comment]
If you're interested, you might want to read our development blog on how we addressed CC licenses and their display through the plugin we made.

https://beeldengeluid.github.io/extending-peertube/category/cc-plugin.html

**Comment by Poslovitch** [comment]
Well, that plugin should be part of the core I think.

**Comment by Maniues** [comment]
@Poslovitch You're right. But actually, peertube should has at least license versions captions. Can you add them?

**Comment by Maniues** [comment]
@frankstrater it's interesting, but it's written in Python, but PeerTube is written in TS. Is it possible to integrate this plugin with PeerTube?

**Comment by frankstrater** [comment]
> @frankstrater it's interesting, but it's written in Python, but PeerTube is written in TS. Is it possible to integrate this plugin with PeerTube?

The CC plugin was part of a bigger project that contains some other "tools" (Python scripts). The plugin is written in Typescript, published on npmjs and available for install through the PeerTube admin interface. See: 

https://www.npmjs.com/package/peertube-plugin-creative-commons

**Comment by Maniues** [comment]
Oh, yes, thanks! So, python scripts are to use by user to use with PeerTube, not to install on instance of PeerTube?

**Comment by frankstrater** [comment]
> Oh, yes, thanks! So, python scripts are to use by user to use with PeerTube, not to install on instance of PeerTube?

Those Python script do stuff with the API of your instance (like bulk importing and/or editing videos). Those are not part of the PeerTube codebase, but more like 'extensions' (hence the project name Extending PeerTube)

**Comment by Maniues** [comment]
So, I can download them and use with my instance, but CC Plugin is designed to integrate it with PeerTube codebase, right?

**Comment by frankstrater** [comment]
Yes, that's right.

**Comment by sorairolake** [comment]
Is this work in progress?

**Comment by frankstrater** [comment]
> Is this work in progress?

It's an open issue, but I doubt anyone is working on content licensing in PeerTube. If there are any, it's probably outside the PeerTube core, so maybe someone is working on an alternative license plugin.

**Comment by valerio-bozzolan** [comment]
Could anybody be so kind to share the link of the file which is responsible for rendering that license badge?

I think it could at least help newcomers to improve the default label under the video.

🟠 CURRENT:

| License | [Attribution - Share Alike](https://videos.pair2jeux.tube/search?licenceOneOf=2) |
|---------|------|

🟢 DESIRED:

| License | [Attribution - Share Alike](https://creativecommons.org/licenses/by-sa/4.0/) · [🔍](https://videos.pair2jeux.tube/search?licenceOneOf=2) |
|---------|------|

So, the proposal is:

- 🟢 The license link points to the license version (e.g. https://creativecommons.org/licenses/by-sa/4.0/ )
- 🟢 The license link has the correct metadata such as `<a rel="license"` for SEO reasons
- 🟢 The current link to "similar videos" is still available, but moved to a nearby search icon

This is really a minimal change but quite necessary to easy adoption on Wikimedia Projects, or to whatever other community project who strictly respects copyright, but don't know how to do it, if the copyright info is not specified correctly.

**Comment by valerio-bozzolan** [comment]
> Could anybody be so kind to share the link of the file which is responsible for rendering that license badge?

Anybody has a vague idea? :)

**Comment by frankstrater** [comment]
> > Could anybody be so kind to share the link of the file which is responsible for rendering that license badge?
> 
> Anybody has a vague idea? :)

https://github.com/beeldengeluid/peertube-plugin-creative-commons/blob/main/client/video-watch-client-plugin.js

@Chocobozzz Would you consider improving the license model in PeerTube core by adding license deed URLs in constants? So instead of the array (keys 1 to 9) with license name as values, make it a multi-array with license url for each key?

**Comment by Chocobozzz** [comment]
Sure!
